cql3/expr/expression.cc

```diff
@@ -965,11 +1055,84 @@ std::ostream& operator<<(std::ostream& os, const expression& expr) {
                 }, fc.func);
             },
             [&] (const cast& c)  {
-                fmt::print(os, "({} AS {})", *c.arg, c.type);
+                std::visit(overloaded_functor{
+                    [&] (const cql3_type& t) {
+                        fmt::print(os, "({} AS {})", *c.arg, t);
+                    },
+                    [&] (const shared_ptr<cql3_type::raw>& t) {
+                        fmt::print(os, "({}) {}", t, *c.arg);
+                    },
+                }, c.type);
             },
             [&] (const field_selection& fs)  {
                 fmt::print(os, "({}.{})", *fs.structure, fs.field);
             },
+            [&] (const null&) {
+                // FIXME: adjust tests and change to NULL
+                fmt::print(os, "null");
+            },
+            [&] (const bind_variable&) {
+                // FIXME: store and present bind variable name
+                fmt::print(os, "?");
```

Comment:
Any changes to the string form trigger failures in unit tests, so these need to done carefully and outside this large change.

As an example, printing null as NULL causes tests to fail.

:{bind_index} is wrong, since it means we can't round-trip CQL.

Reply:
OK.